### PR TITLE
better error handler

### PR DESCRIPTION
**Pull request by KevinMRmz**
Better way to handle errors

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -5,3 +5,5 @@ PORT=8000
 
 JWT_SECRET=@McQfTjWnZr4u7x!A%D*F-JaNdRgUkXp
 TOKEN_DURATION=30d
+
+MAX_PATIENTS_FOR_NURSE=6
```
Comment (JavierRuvalcaba) [inline .env:9]: Remove this file and add a `.env.example` instead.
The content of the `.env.example` should be like this: 

```
JWT_SECRET=CHANGE_ME
TOKEN_DURATION=CHANGE_ME
MAX_PATIENTS_FOR_NURSE=CHANGE_ME
```

File: services/patient-service/patient-service.js [hunk 2]
```diff
@@ -52,7 +52,18 @@ class PatientService {
...
   async returnNursePatients(idNurse) {
+    if (!idNurse) {
+      throw new BadRequestError("Please provide an Employee id");
+    }
+
+    const employee = await EmployeeModel.findById(idNurse);
+
+    if (!employee) {
+      throw new NotFoundError("Employee was not found");
+    }
```
Comment (JavierRuvalcaba) [inline services/patient-service/patient-service.js]: We can be more specific. The function is to return the patients of a nurse, so change `employee` word to `nurse`

```suggestion
    if (!idNurse) {
      throw new BadRequestError("Please provide a Nurse id");
    }

    const nurse = await EmployeeModel.findById(idNurse);

    if (!nurse) {
      throw new NotFoundError("Nurse was not found");
    }
```

File: services/patient-service/patient-service.js [hunk 3]
```diff
@@ -142,9 +153,23 @@ class PatientService {
...
+
+    const employee = await EmployeeModel.findById(idNurse);
+
+    if (employee.role !== "NURSE") {
+      throw new BadRequestError(
+        "The Employee you're trying to set is not a nurse"
+      );
+    }
+
```
Comment (JavierRuvalcaba) [inline services/patient-service/patient-service.js]: Validate first if the nurse exists or if the idNurse correspond to a nurse. After that validate if the nurse has already 6 patients or not.

```suggestion
    const employee = await EmployeeModel.findById(idNurse);

    if (employee.role !== "NURSE") {
      throw new BadRequestError(
        "The Employee you're trying to set is not a nurse"
      );
    }
    
    const patients = await this.returnNursePatients(idNurse);

    if (patients.length >= parseInt(process.env.MAX_PATIENTS_FOR_NURSE)) {
      throw new BadRequestError("This nurse can only have 6 patients or less");
    }
```

File: services/record-service/notifications-service.js [hunk 4]
```diff
@@ -55,16 +69,64 @@ class NotificationService {
    * @return {Array<NotificationObject>} All documents belong to a doctor
    */
   async returnDoctorNotifications(idDoctor) {
+    if (!idDoctor) {
+      throw new BadRequestError("Please provide an Doctor ID");
```
Comment (JavierRuvalcaba) [inline services/record-service/notifications-service.js]:
```suggestion
      throw new BadRequestError("Please provide a Doctor ID");
```

File: services/record-service/notifications-service.js [hunk 5]
```diff
@@ -55,16 +69,64 @@ class NotificationService {
...
   }
 
+  /**
+   *
+   * @param {mongoose.Types.ObjectId} idDoctor
+   * @return {Array<NotificationObject>} All notifications matched to a doctor id
+   */
+  async returnAllNewDoctorNotifications(idDoctor) {
+    if (!idDoctor) {
+      throw new BadRequestError("Please provide an Doctor ID");
```
Comment (JavierRuvalcaba) [inline services/record-service/notifications-service.js]:
```suggestion
      throw new BadRequestError("Please provide a Doctor ID");
```

File: services/record-service/notifications-service.js [hunk 6]
```diff
@@ -55,16 +69,64 @@ class NotificationService {
...
+  }
+
   /**
    *
    * @param {mongoose.Types.ObjectId} idTransmitter
    * @return {Array<NotificationObject>} All documents belong to a nurse
    */
   async returnNurseNotifications(idTransmitter) {
+    if (!idTransmitter) {
+      throw new BadRequestError("Please provide a Id for transmitter");
```
Comment (JavierRuvalcaba) [inline services/record-service/notifications-service.js]:
```suggestion
      throw new BadRequestError("Please provide a Nurse Id");
```

**Review by JavierRuvalcaba — CHANGES_REQUESTED** [review]

**Review by pedro123-67we — DISMISSED** [review]
good job!!!!

File: services/patient-service/patient-service.js [hunk 7]
```diff
@@ -52,7 +52,18 @@ class PatientService {
    * @returns {Array<PatientObject>} Patients matched to the nurse ID
    */
   async returnNursePatients(idNurse) {
+    if (!idNurse) {
+      throw new BadRequestError("Please provide an Nurse id");
```
Comment (JavierRuvalcaba) [inline services/patient-service/patient-service.js]:
```suggestion
      throw new BadRequestError("Please provide a Nurse id");
```

**Review by JavierRuvalcaba — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (JavierRuvalcaba) [reply .env:9]: Also remove this file from the repo.

**Review by JavierRuvalcaba — CHANGES_REQUESTED** [review]

**Review by JavierRuvalcaba — APPROVED** [review]

**Review by pedro123-67we — APPROVED** [review]

**Pull request merged**